Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/jupyter_cache/executors/utils.py", line 56, in single_nb_execution
    record_timing=False,
  File "/usr/local/lib/python3.7/dist-packages/nbclient/client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/usr/local/lib/python3.7/dist-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/usr/local/lib/python3.7/dist-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/usr/lib/python3.7/asyncio/base_events.py", line 587, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.7/dist-packages/nbclient/client.py", line 664, in async_execute
    cell, index, execution_count=self.code_cells_executed + 1
  File "/usr/local/lib/python3.7/dist-packages/nbclient/client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/usr/local/lib/python3.7/dist-packages/nbclient/client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# data_tes = np.array(labels)
k_means = KMeans(n_clusters=3, n_init=1, max_iter=1)
k_means.fit(X)
# kmeans.labels_
label = kmeans.predict(X)

centroid = kmeans.cluster_centers_
------------------

[0;31m[0m
[0;31mNameError[0mTraceback (most recent call last)
[0;32m<ipython-input-9-7c59e4005037>[0m in [0;36m<module>[0;34m[0m
[1;32m      1[0m [0;31m# data_tes = np.array(labels)[0m[0;34m[0m[0;34m[0m[0;34m[0m[0m
[1;32m      2[0m [0mk_means[0m [0;34m=[0m [0mKMeans[0m[0;34m([0m[0mn_clusters[0m[0;34m=[0m[0;36m3[0m[0;34m,[0m [0mn_init[0m[0;34m=[0m[0;36m1[0m[0;34m,[0m [0mmax_iter[0m[0;34m=[0m[0;36m1[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0;32m----> 3[0;31m [0mk_means[0m[0;34m.[0m[0mfit[0m[0;34m([0m[0mX[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m      4[0m [0;31m# kmeans.labels_[0m[0;34m[0m[0;34m[0m[0;34m[0m[0m
[1;32m      5[0m [0mlabel[0m [0;34m=[0m [0mkmeans[0m[0;34m.[0m[0mpredict[0m[0;34m([0m[0mX[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m

[0;31mNameError[0m: name 'X' is not defined
NameError: name 'X' is not defined

